src/api/common/schemas/get-paths.json

```diff
@@ -6,7 +6,23 @@
     "type": "object",
     "properties": {
       "source": {"$ref": "adjustment"},
-      "destination": {"$ref": "adjustment"},
+      "destination": {
+        "type": "object",
```

Comment:
By "correctly" I mean it should be the actual counterparty, not any special case. The only option I see is that we could throw a validation error if the destination counterparty is the destination address.